apps/playground/src/composables/useStore.ts

```diff
@@ -50,16 +52,32 @@ export const useStore = () => {
       dependencyVersion: computed(() => {
         // the dependencyVersion must be a real version number and not a range like "alpha"
         const version =
-          onyxVersion.value.includes(".") || !availableVersions.value.length
+          onyxVersion.value.includes(".") || !availableOnyxVersions.value.length
             ? onyxVersion.value
-            : availableVersions.value[0];
+            : availableOnyxVersions.value[0];
         return { "sit-onyx": version };
       }),
     },
     // initialize repl with previously serialized state
     location.hash.slice(1),
   );
 
+  watch(onyxVersion, (newVersion) => {
+    updateQueryParam("onyxVersion", newVersion);
+  });
+  watch(
+    () => store.typescriptVersion,
+    (newVersion) => {
+      updateQueryParam("typescriptVersion", newVersion);
+    },
+  );
+
+  const updateQueryParam = (key: string, value: string) => {
+    const url = new URL(location.href);
```

Comment:
should there be an early return if the same version is set again, the URL/history does not need to update?
e.g. our select emits update:modelValue even when you click the current value again, or maybe it's even to prevent the possible loop mentioned in the other comment.
![image](https://github.com/user-attachments/assets/8af44a37-7ba2-411c-a0a8-abd5ee7f7ea3)

Reply:
Vue will already take care of this so if you have a ref with value "beta" and you change it to "beta", no watcher or reactive updates will be triggered